internal/armada/server.go

```diff
@@ -53,12 +67,36 @@ func Serve(config *configuration.ArmadaConfig, healthChecks *health.MultiChecker
 	authServices := auth.ConfigureAuth(config.Auth)
 	grpcServer := grpcCommon.CreateGrpcServer(authServices)
 
-	// Allows for registering functions to be run periodically in the background
+	// Shut down grpcServer if the context is cancelled.
+	// Give the server 5 seconds to shut down gracefully.
+	services = append(services, func() error {
+		<-ctx.Done()
+		go func() {
+			time.Sleep(5 * time.Second)
```

Comment:
Not something to change. However we should probably review the graceful shutdown times and make sure they are sensible and consistent (and configurable?)

Reply:
I agree.